session-replacement/src/main/java/com/amadeus/session/SessionManager.java

```diff
@@ -263,19 +264,27 @@ public RepositoryBackedSession getSession(RequestWithSession request, boolean cr
     }
     if (session == null && create) {
       id = forceId != null ? forceId : tracking.newId();
+      putIdInLoggingMdc(id);
       logger.info("Creating new session with sessionId: '{}'", id);
       session = newSession(id);
     }
     if (session != null) {
       session.checkUsedAndLock();
-      // If logging has MDC, we add session id to MDC under configured key.
-      if (configuration.isLoggingMdcActive()) {
-        MDC.put(configuration.getLoggingMdcKey(), session.getId());
-      }
     }
     return session;
   }
 
+  // If logging has MDC, we add session id to MDC under configured key.
```

Comment:
I would put a Javadoc comment instead of an inline comment, to be consistent with what we did for the other methods of this class.
We can also precise that we only put the Id if not null